src/VisualStudio/Core/Def/CodeCleanup/AbstractCodeCleanUpFixer.cs

```diff
@@ -193,11 +192,120 @@ private async Task<bool> FixAsync(
+                            .Concat(projectChanges.GetChangedAdditionalDocuments())
+                            .Concat(projectChanges.GetChangedAnalyzerConfigDocuments()).ToImmutableArray();
+
+                        if (changedDocuments.Length == 0)
+                        {
+                            //nothing has changed in this project. We can't apply this change.
+                            return false;
+                        }
+
+                        foreach (var documentId in changedDocuments)
+                        {
+                            var originalDocument = projectChanges.NewProject.GetRequiredDocument(documentId);
+                            var currentDocument = currentSolution.GetTextDocument(documentId);
+
+                            if (currentDocument is null)
+                            {
+                                // Document is gone, we can't apply this change.
+
+                                return false;
+                            }
+
+                            if (originalDocument.HasTextChanged(currentDocument, ignoreUnchangeableDocument: false))
+                            {
+                                // Document has changed, we can't apply this change.
+                                return false;
+                            }
+                        }
+                    }
+
+                    // If we got here, all documents in this project are the same as they were when we started.
+                    var codeCleanUpChanges = solution.GetChanges(currentSolution);
+
+                    foreach (var projectChanges in codeCleanUpChanges.GetProjectChanges())
+                    {
+                        var codeCleanupDocChanges = projectChanges.GetChangedDocuments()
+                            .Concat(projectChanges.GetChangedAdditionalDocuments())
+                            .Concat(projectChanges.GetChangedAnalyzerConfigDocuments()).ToImmutableArray();
+
+                        if (codeCleanupDocChanges.Length == 0)
+                        {
+                            //nothing has changed in this project. We can't apply this change.
+                            return false;
+                        }
+
+                        foreach (var documentId in codeCleanupDocChanges)
+                        {
+                            var changedDocument = projectChanges.NewProject.GetRequiredDocument(documentId);
+                            var currentDocument = solution.GetRequiredDocument(documentId);
+
+                            if (currentDocument is null)
+                            {
+                                // Document is gone, we can't apply this change.
+                                return false;
+                            }
+
+                            solution = currentSolution.WithTextDocumentText(documentId, changedDocument.GetTextSynchronously(cancellationToken));
```

Comment:
We should use `updatedSolution` here. We will keep modifying it so that it accumulates all the changes from `changedSolution`.

```suggestion
                            updatedSolution = updatedSolution.WithTextDocumentText(documentId, changedDocument.GetTextSynchronously(cancellationToken));
```